R/get_source_expressions.R

```diff
@@ -483,19 +483,10 @@ get_single_source_expression <- function(loc,
 get_source_expression <- function(source_expression, error = identity) {
   parse_error <- FALSE
 
-  tryCatch(
-    source_expression$parsed_content <- parse(
-      text = source_expression$content,
-      srcfile = source_expression,
-      keep.source = TRUE
-    ),
-    error = error
-  )
-
-  # This needs to be done twice to avoid
-  #   https://bugs.r-project.org/bugzilla/show_bug.cgi?id=16041
+  # This object should be named `e` because, if invalid characters are present,
+  # the error needs to be assigned in the parent frame, where it is named `e`.
```

Comment:
For this reason, the content can't be assigned to `parsed_content` or some other, more informative name.

Reply:
isn't the parent's name given by the first argument to assign()?

so assign("e", parsed_content, ...) works?

Reply:
Ah, yes, true that.

Sorry, I was doing search and replace, which changed both `e` and `"e"`, and that's why it wasn't working.

I have made the necessary change now and the object is called `parsed_content`. 

Thanks!